frontend-public/src/utils/actionlessNetworkCalls.js

```diff
@@ -2,21 +2,22 @@ import axios from "axios";
 import { notification } from "antd";
 import { createRequestHeader } from "@/utils/RequestHeaders";
 import { ENVIRONMENT } from "@/constants/environment";
-import { DOCUMENT_MANAGER_FILE_GET_URL } from "@/constants/API";
+import { DOCUMENT_MANAGER_FILE_GET_URL, DOCUMENT_MANAGER_TOKEN_GET_URL } from "@/constants/API";
 
-const fileDownload = require("js-file-download");
-
-const downloadFileFromDocumentManager = (docManagerGuid, filename) => {
-  if (!docManagerGuid || !filename) {
-    throw new Error("Must provide both docManagerGuid and filename");
+const downloadFileFromDocumentManager = (docManagerGuid) => {
+  if (!docManagerGuid) {
+    throw new Error("Must provide docManagerGuid");
   }
 
   // TODO: Update url when Document Manager moves to its own microservice.
-  const url = `${ENVIRONMENT.apiUrl + DOCUMENT_MANAGER_FILE_GET_URL}/${docManagerGuid}`;
   axios
-    .get(url, { responseType: "arraybuffer", ...createRequestHeader() })
+    .get(
+      `${ENVIRONMENT.apiUrl + DOCUMENT_MANAGER_TOKEN_GET_URL(docManagerGuid)}`,
+      createRequestHeader()
+    )
     .then((response) => {
-      fileDownload(response.data, filename);
+      const token = { token: response.data.token_guid };
+      window.location = `${ENVIRONMENT.apiUrl + DOCUMENT_MANAGER_FILE_GET_URL(token)}`;
```

Comment:
I'm concerned about us setting `window.location`, as I think it has effects on our router behaviour.

It looks like this is to trigger the document to download? I'd suggest confirming that the back button still works after downloading a document.

Reply:
It doesn't actually navigate away, surprisingly. Since the backend response sets a download header, the browser magically understands that its supposed to download the file.

This is the first time iv'e ever seen where the browser's builtin behavior is 100% what we wanted.

Reply:
Fair enough! If it's doing what we want and not causing any issues with React router then I'm happy. I'll mark this as resolved